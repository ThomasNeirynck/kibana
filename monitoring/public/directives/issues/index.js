define(function (require) {
  var _ = require('lodash');
  var template = require('plugins/monitoring/directives/issues/index.html');
  var module = require('ui/modules').get('monitoring/directives', []);
  module.directive('monitoringIssues', function (monitoringMetrics) {
    return {
      restrict: 'E',
      scope: {
        title: '@',
        source: '=',
        link: '@'
      },
      template: template,
      link: function ($scope) {
        $scope.$watch('source.data', function () {
          $scope.status = '';
          if (_.some($scope.source.data, { status: 'yellow'})) {
            $scope.status = 'yellow';
          }
          if (_.some($scope.source.data, { status: 'red' })) {
            $scope.status = 'red';
          }
          $scope.total = $scope.source.data.length;
          $scope.displaying = ($scope.total <= 6) ? $scope.total : 6;
        });
      }
    };
  });
});


