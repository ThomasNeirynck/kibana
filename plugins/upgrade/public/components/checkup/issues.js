import classNames from 'classnames';
import React from 'react';

import { Issue } from './issue';


export function Issues({ className, issues }) {
  const classes = classNames('kuiMenu kuiMenu--contained', className);

  return (
    <ul className={ classes }>
      { issues.map((issue, issueIndex) => (
        <Issue
          details={ issue.details }
          key={ issueIndex }
          level={ issue.level }
          message={ issue.message }
          url={ issue.url }
        />
      )) }
    </ul>
  );
}

Issues.propTypes = {
  className: React.PropTypes.string,
  issues: React.PropTypes.array,
};

Issues.defaultProps = {
  className: null,
  issues: [],
};
