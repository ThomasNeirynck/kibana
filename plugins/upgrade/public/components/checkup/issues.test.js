import { shallow } from 'enzyme';
import React from 'react';

import { Issues } from './issues';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('Issues', () => {
  test('renders the className property', () => {
    const component = (
      <Issues className="myClass" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a list of Issue components', () => {
    const issues = [
      {
        details: 'issue 1 details',
        level: 'issue 1 level',
        message: 'issue 1 message',
        url: 'issue 1 url',
      },
      {
        details: 'issue 2 details',
        level: 'issue 2 level',
        message: 'issue 2 message',
        url: 'issue 2 url',
      },
    ];
    const component = (
      <Issues issues={ issues } />
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
