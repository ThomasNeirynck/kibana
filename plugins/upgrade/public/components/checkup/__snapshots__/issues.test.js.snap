// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issues renders a list of Issue components 1`] = `
<ul
  className="kuiMenu kuiMenu--contained"
>
  <Issue
    details="issue 1 details"
    issueInfo={null}
    level="issue 1 level"
    message="issue 1 message"
    url="issue 1 url"
  />
  <Issue
    details="issue 2 details"
    issueInfo={null}
    level="issue 2 level"
    message="issue 2 message"
    url="issue 2 url"
  />
</ul>
`;

exports[`Issues renders the className property 1`] = `
<ul
  className="kuiMenu kuiMenu--contained myClass"
/>
`;
